components/load_from_hf_hub/fondant_component.yaml

```diff
@@ -19,11 +19,11 @@ args:
     description: Optional argument, a list containing the original image column names in case the 
       dataset on the hub contains them. Used to format the image from HF hub format to a byte string.
     type: list
-    default: None
+    default: []
   n_rows_to_load:
     description: Optional argument that defines the number of rows to load. Useful for testing pipeline runs on a small scale
     type: int
-    default: None
+    default: -1
   index_column:
     description: Column to set index to in the load component, if not specified a default globally unique index will be set
     type: str
```

Comment:
Does it make sense to later on revert back to None and handle the conversion internally? It could be something like setting an arbitrary value if it's None in the fondant component and parsing it back to None again when the component is run

Reply:
I'm not sure if it's such a bad thing that the default needs to match the argument type. We would also need an arbitrary value for each type to make this work.

Reply:
Before we would assign a value as an optional argument if the default is set to `None`. Now the notion of optional arguments is missing and optional arguments are checked implicitly against arbitrary values defined in the default which can be a bit confusing. 

Was more thinking from the perspective of the end user. So if they would assign a default to `None`, we would handle it as optional and return it as `None` to the end user. Implementing this is still not very clear to me but we could just assign arbitrary known values. For example, if the users sets an list argument to `None`, we would pass it as `[]` and the convert it back to `None` during parsing